From e8b9d5ca5ebed5f2e076e6e6c975ec37efc21c5b Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Fri, 8 Jun 2018 12:42:49 +0200
Subject: [PATCH 1/1] build: Unbreak the icon installation in gnome-3-28

The icons were getting directly copied to $data_dir/icons without
retaining the directory hierarchy inside.

Fallout from d2410d0959094dee8cc3e1276b255e8fb991fe31

It was also broken in master, but was fixed by
c115f5bfb56aa9fe42356c5f4f9ee87f6c87f454 and
578707e9924c32a808e12c2830f18156ccb109f6

https://gitlab.gnome.org/GNOME/gnome-boxes/issues/217
---
 data/meson.build | 26 +-------------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 847734e5..9f1c4b48 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,28 +1,4 @@
-icondir = join_paths(data_dir, 'icons')
-
-install_data('icons/hicolor/16x16/apps/org.gnome.Boxes.png',
-  install_dir: icondir,
-)
-
-install_data('icons/hicolor/24x24/apps/org.gnome.Boxes.png',
-  install_dir: icondir,
-)
-
-install_data('icons/hicolor/symbolic/apps/org.gnome.Boxes-symbolic.svg',
-  install_dir: icondir,
-)
-install_data('icons/hicolor/32x32/apps/org.gnome.Boxes.png',
-  install_dir: icondir,
-)
-
-install_data('icons/hicolor/48x48/apps/org.gnome.Boxes.png',
-  install_dir: icondir,
-)
-
-install_data('icons/hicolor/256x256/apps/org.gnome.Boxes.png',
-  install_dir: icondir,
-)
-
+install_subdir('icons/hicolor/', install_dir: join_paths (data_dir, 'icons'))
 
 resource_files = files ('gnome-boxes.gresource.xml')
 resources = gnome.compile_resources ('org.gnome.Boxes',
-- 
2.18.0

